src/server/zset_family.cc

```diff
@@ -1058,6 +1058,38 @@ struct SetOpArgs {
   bool with_scores = false;
 };
 
+void HandleOpStatus(ConnectionContext* cntx, OpStatus op_status) {
+  switch (op_status) {
+    case OpStatus::INVALID_FLOAT:
+      return (*cntx)->SendError("weight value is not a float", kSyntaxErrType);
+    default:
+      return (*cntx)->SendError(op_status);
+  }
+}
+
+OpResult<ScoredMap> IntersectResults(const vector<OpResult<ScoredMap>>& results, AggType agg_type) {
+  OpResult<ScoredMap> result;
+  for (auto op_res : results) {
+    if (op_res.status() == OpStatus::SKIPPED)
+      continue;
+
+    if (!op_res) {
+      return op_res.status();
+    }
+
+    if (op_res->empty()) {
+      return op_res.status();
```

Comment:
I think it would be more readable to return an empty `ScoredMap` here, instead of an implied OK status

```suggestion
      return ScoredMap{};
```